### ISSUE TITLE
Download history + skipping of already downloaded images

### ISSUE THREAD

**Issue by TestPolygon** [ccloli/E-Hentai-Downloader#177]
I would like to have on option to hold the download history. And download images only once. Do not download the image again when I download the other gallery containing this image (even the image has the other name).

Each image has ID (the first 10 bytes of SHA-1), you can see it in URLs: `.../s/{image-id}/{gallery-id}-{position}`. The same images have the same `{image-id}`, only `{gallery-id}` and `{position}` are differed. _(note: the file name may be different too)_

You can store IDs of already downloaded images to implement an option that with enabled one the userscript will download only images that was not downloaded before.

Also the skipped images should be noted in `info.txt`.

**Comment by TestPolygon** [comment]
It would be very useful for people why want to download all images by some author tag. No unnecessary requests, less downloaded data (because of there is no duplicates). Easilier to download a gallery updates.

**Comment by TestPolygon** [comment]
You can also mark downloaded images in galleries with colored border. I think it would be useful too.

**Comment by ccloli** [comment]
Yes, the id is the first 10 letters of the file SHA-1, however it may not accure enough. Just now I noticed the thumb url contains full SHA-1.

For example, the thumbnail url of `https://e-hentai.org/s/e91885fc7e/1809948-57`
is `https://ehgt.org/t/e9/18/e91885fc7ec0212a975ef371d9c9816c3f3807b0-768879-884-1250-jpg_l.jpg`.

The first 10 letters of file name part of thumbnail url is exact as the page url SHA-1 part `e91885fc7e`, so we can determine the format of thumbnail url is
`https://${domain}/t/${sha1.substr(0, 2)}/${sha1.substr(2, 2)}/${sha1}-${size}-${width}-${height}-${format}_${thumbsize}.jpg`.

**Comment by TestPolygon** [comment]
I don't know how the site handle the sitiation when the different images have the hashes with the same first five bytes. _Probably_ it gives a unical ID, or maybe not.

It does not seem critical to work only with the first 5 bytes (10 hex characters, or 40 bites). 5 bytes is enough to give IDs for 1099511627776 items (`2**40`, or `16**10`, or `256**5`).
Not sure, is [Birthday attack](https://en.wikipedia.org/wiki/Birthday_attack) can be applied to this case, but with counting of this `2**20` is 1048576 – not great, not terrible.

But it looks OK, if this feature would work with only enabled "Large" (not "Normal") thumbnails if you want to store the full hashes.
